src/packet/packetlist.js

```diff
@@ -18,7 +18,10 @@ import defaultConfig from '../config';
  * @returns {Object} New packet object with type based on tag
  * @throws {Error|UnsupportedError} for disallowed or unknown packets
  */
-export function newPacketFromTag(tag, allowedPackets) {
+export function newPacketFromTag(tag, allowedPackets, config = defaultConfig) {
+  if (config.additionalAllowedPackets && config.additionalAllowedPackets.length) {
+    allowedPackets = { ...allowedPackets, ...util.constructAllowedPackets(config.additionalAllowedPackets) };
```

Comment:
This function is now called for every packet (if the config option is set); since `util.constructAllowedPackets` is already exported, I would just make the application call that, and pass the returned object in the config. That way, the object can also be shared across calls to OpenPGP.js.

Reply:
I was also wondering where to put the logic. We could also move it up the call stack into `PacketList.read` or `PacketList.fromBinary` to reduce the number of calls.

I find additionalAllowedPackets as an Array easier to document, but if you prefer the object map here then I can change that as well. But I don't see that `util.constructAllowedPackets` is exported and util seems to be a private module?

Reply:
Oh, you're right, I forgot we unexported `util`. Yeah, then let's put it in `PacketList.read`?